src/Corax/Querying/IndexSearcher.cs

```diff
@@ -393,8 +393,10 @@ public List<string> GetFields()
     public FieldIndexingMode GetFieldIndexingModeForDynamic(Slice name)
     {
         _persistedDynamicTreeAnalyzer ??= _transaction.ReadTree(Constants.IndexWriter.DynamicFieldsAnalyzersSlice);
-        var readResult = _persistedDynamicTreeAnalyzer?.Read(name);
-        if (readResult == null)
+        if (_persistedDynamicTreeAnalyzer == null)
+            return FieldIndexingMode.Normal; 
+        var readResult = _persistedDynamicTreeAnalyzer.Read(name);
+        if (!readResult.HasValue)
```

Comment:
We rather avoid the usage of `!` in conditional checks and prefer ` == false` then. It's sometimes easy to overlook `!`